src/ng-select/lib/selection-model.ts

```diff
@@ -62,14 +64,25 @@ export class DefaultSelectionModel implements SelectionModel {
     }
 
     private _setChildrenSelectedState(children: NgOption[], selected: boolean) {
-        children.forEach(x => x.selected = selected);
+        children.forEach(x => {
+            if (x.disabled !== true) {
+                x.selected = selected;
+            }
+        });
     }
 
     private _removeChildren(parent: NgOption) {
-        this._selected = this._selected.filter(x => x.parent !== parent);
+        this._selected = [
```

Comment:
Yep, fixed.

Reply:
Ah nevermind, I realized again that this is require when the `selectableGroupAsModel = true`. The change you propose would break this unit test:

```
it('should not unselect disabled items within a group', () =>
  cmp.groupBy = 'groupKey';
  list.setItems([
     { label: 'K1', val: 'V1', groupKey: 'G1' },
     { label: 'K2', val: 'V2', groupKey: 'G1', disabled: true },
   ]);

  list.select(list.items[2]); // K2
  list.select(list.items[0]); // G1
  expect(list.selectedItems.length).toBe(1);
  expect(list.selectedItems[0].label).toBe('G1');

  list.unselect(list.items[0]); // G1
  expect(list.selectedItems.length).toBe(1);
  expect(list.selectedItems[0].label).toBe('K2');
});
```

Reply:
I simplified the code.

Reply:
Update: I added a new check on the second filter to ensure it will only add the selected disabled items for that parent:

`...parent.children.filter(x => x.parent === parent && x.disabled && x.selected)`

I updated the unit tests to cover this case.